tensorboard/webapp/runs/views/runs_table/runs_table_component.ng.html

```diff
@@ -27,186 +27,125 @@
   </div>
 </div>
 <div class="table-container">
-  <table
-    mat-table
-    [dataSource]="dataSource"
+  <div role="grid">
+    <ng-container *ngTemplateOutlet="header"></ng-container>
+    <ng-container *ngTemplateOutlet="selectAll"></ng-container>
+    <ng-container *ngFor="let item of pageItems; trackBy: tableTrackBy">
+      <ng-container *ngTemplateOutlet="row; context:{item: item}"></ng-container
+    ></ng-container>
+  </div>
+
+  <div *ngIf="loading" class="loading">
+    <mat-spinner mode="indeterminate" diameter="28"></mat-spinner>
+  </div>
+
+  <div *ngIf="!loading && allItemsLength === 0" class="no-runs">No Runs</div>
+
+  <div
+    *ngIf="!loading && allItemsLength > 0 && filteredItemsLength === 0"
+    class="no-runs"
+  >
+    <span>No runs match "<code>{{ regexFilter }}</code>"</span>
+  </div>
+</div>
+
+<mat-paginator
+  *ngIf="usePagination"
+  [pageSizeOptions]="[5, 10, 20]"
+  [pageIndex]="paginationOption.pageIndex"
+  [pageSize]="paginationOption.pageSize"
+  [length]="filteredItemsLength"
+  showFirstLastButtons
+  (page)="onPaginationChange.emit($event)"
+></mat-paginator>
+
+<ng-template #selectAll>
+  <div
+    *ngIf="allPageItemsSelected() && numSelectedItems !== allItemsLength"
+    class="select-all"
+    role="rowheader"
+  >
+    <span
+      >All runs in this page are selected but not all runs ({{ numSelectedItems
+      }} of {{ allItemsLength }}) are selected.</span
+    ><button mat-button (click)="onSelectAllPages.emit()">
+      Select all runs
+    </button>
+  </div>
+</ng-template>
+
+<ng-template #header>
+  <div
+    class="header"
     matSort
     (matSortChange)="handleSortChange($event)"
     [matSortActive]="sortOption.column"
-    [matSortDirection]="sortOption.direction"
-    [trackBy]="tableTrackBy"
+    role="rowheader"
   >
-    <!-- Note that matColumnDef has to be static and cannot be bound to enum -->
-    <ng-container matColumnDef="checkbox">
-      <th mat-header-cell *matHeaderCellDef>
-        <mat-checkbox
-          [checked]="allPageItemsSelected()"
-          [indeterminate]="!allPageItemsSelected() && somePageItemsSelected()"
-          (change)="handlePageToggle()"
-        ></mat-checkbox>
-      </th>
-      <td mat-cell *matCellDef="let item">
-        <mat-checkbox
-          [checked]="item.selected"
-          (change)="onSelectionToggle.emit(item)"
-        ></mat-checkbox>
-      </td>
-    </ng-container>
+    <span
+      *ngFor="let columnId of columns"
+      role="cell"
+      [ngClass]="['column', columnId]"
```

Comment:
It is used in the existing code. It is used in CSS and it is used in the test. I will _not_ drop it in this PR since DOM structure refactor is the major point, fixing for a potential bug.

Reply:
I mis-remembered the code I wrote some time ago. You are right. This `columnId` business is a new addition and is now removed.